### ISSUE TITLE
About the perfermance of results and sampling

### ISSUE THREAD

**Issue by gangli95** [josedolz/LiviaNET#10]
Hi, Jose. Thanks for your guide and I had run the example code successfully using the given data, but the result was not satisfactory. Here is the Testing result I got:
![given_data](https://user-images.githubusercontent.com/33647715/33377436-6fdcf418-d54c-11e7-9d3b-e9fc67f8de2c.jpg)
And this is the result using my own data:
![own_data](https://user-images.githubusercontent.com/33647715/33377437-70134798-d54c-11e7-9f18-14ea83b82d7a.jpg)

Should the results be like these above using the example code ? Or there is something wrong in my training?

Another question is about sampling. In your paper, it said that the segment size is 27*27*27, and at each subepoch, a total of 500 samples were randomly selected from the training image segments.  Is the segment size equals to sample size? (I didn't find anything about segment in your code, just sampling[25*25*25] ).  And the 500 samples should be selected from several MR/GT images, it's that right?

**Comment by gangli95** [comment]
BTW,  is the architecture in the code different from that in the paper?

**Comment by josedolz** [comment]
Hi @gangli95 

Thanks for you comments. So, few things about this.

1 - For the sampling. Segments, it is just another way to call the sub-patches. So segment size is the same than sample size. It is actually 27x27x27 for training and I think we used 35x35x35 for segmentation in that paper. And yes, the 500 samples must be selected from different (ideally all) subjects during training.

2 - I realized last week that there is a small error in the code (see [this](https://github.com/josedolz/LiviaNET/issues/9) ). Basically I hard-coded the values of the dropout rates for the fully convolutional connected layers, being equal to 0.0. A workaround (that I just made for the following results) might be to manually set these dropout rates to 0.25 in all the fully connected layers (Keep 0.5 for softmax).

3 - The architecture in the config file is a tinny net to simply try the code. It is not the one used in the paper. Just have a look to the paper and you will easily see what you need to change in the config file (number of layers, kernels per layer, interconnected layers,etc...). I just run it with the current code and current data, just modifying the config file properly and the trick of the dropout and only after 3 epochs I got this for the validation subject (training on 6 subjects and validating in only one):

![subcortical](https://user-images.githubusercontent.com/5444841/33402850-eb3867b2-d52b-11e7-8303-97298f9ca933.png)

**Comment by josedolz** [comment]
Just one question. When you use your data? Do you use it for training and testing? Or only for testing? When I started to work with this dataset, I used matlab to load nifti files, which made changes in some axis (That is why the images look weird in your first row). So if you train with the given data, as it is, and test in your own data, you will have high chances that it will not work, since the appearance is way different.

**Comment by gangli95** [comment]
Thanks for your replying!  I used my data for training and testing, and it looks like now there is something wrong  with the pre-processing, which I used the FreeSurfer to do it. \
Could you please give me the new config file that that you used above? I have difficulty in changing there parameters.  Thanks.

**Comment by josedolz** [comment]
Sure,

I just uploaded the config file I used to generate that image. Here is the [link](https://github.com/josedolz/LiviaNET/blob/master/src/LiviaNET_Config_NeuroPaper.ini).

**Comment by gangli95** [comment]
Thanks, Jose. I had run the code using the new config file, and its
perfermance is great!

2017年11月30日星期四，Jose Dolz <notifications@github.com> 写道：

**Comment by josedolz** [comment]
Hi @gangli95 

I am glad you could have reproduce the results. I hope you enjoy it and that it can be useful for your research.

Jose

**Comment by gangli95** [comment]
Hi, @josedolz  I got problems again, the results contained numbers of small regions when I used my data(both training and testing).  And after I looked up at the [https://github.com/josedolz/LiviaNET/issues/3](url),  I did the normalization again, but the results were still similar.  Did I need to do the post-pressing? I thought it was included in your code already.
Here are the result(one subject):

![result1](https://user-images.githubusercontent.com/33647715/33942057-b5c9fe90-e04f-11e7-9228-5d9acbd91b0b.png)

![result2](https://user-images.githubusercontent.com/33647715/33942065-bbab45da-e04f-11e7-9626-9488a6c075c3.png)

**Comment by josedolz** [comment]
Hi @gangli95 I do not see any problem on your results, they look pretty ok. There are some isolated areas, which is normal given the reduced receptive field of the network. As I did in the paper, you need to apply some sort of component analysis to remove those isolated blobs. It is a straightforward step, so I did not include it in the code, since it might vary depending on the application.

**Comment by gangli95** [comment]
Thanks for the reply!  And it's great for me to hear the results are good.  I'll try using regionprop in matlab to deal with. BTW,  I think that [regionprops](https://cn.mathworks.com/help/images/ref/regionprops.html) may cause regions' loss, both I expect and not.  And maybe [regionprops3](https://cn.mathworks.com/help/images/ref/regionprops3.html#responsive_offcanvas) (introduced in MatLAB R2017b) may improve this.
